core/src/com/biglybt/core/internat/MessageText.java

```diff
@@ -866,4 +826,30 @@ public static void revertToDefaultLocale() {
 		Locale	new_locale );
   }
 
+	/**
+	 * @return matching Locale or {@link Locale#ROOT} if unrecognized format or imparsable.
+	 */
+	static Locale parseFormattedLocaleString(String savedLocaleString) {
+		Locale.Builder localeBuilder = new Locale.Builder();
```

Comment:
@TuxPaper I just realized that Locale.Builder has API level 21 although its plain Java 7.
I like the builder because it utilize the Locale#LOCALECACHE.